crates/transaction-pool/src/pool/mod.rs

```diff
@@ -553,6 +568,29 @@ where
         });
     }
 
+    /// Notify all listeners about a blob sidecar for a newly inserted blob (eip4844) transaction.
+    fn on_new_blob_sidecar(&self, tx_hash: &TxHash, sidecar: &BlobTransactionSidecar) {
+        let mut sidecar_listeners = self.blob_transaction_sidecar_listener.lock();
+        let new_blob_event = NewBlobSidecar { tx_hash: *tx_hash, sidecar: sidecar.clone() };
```

Comment:
this is still not optimal, because this still creates a clone regardless.

this must be moved into the closure instead

Reply:
ah ok .. `new_blob_event` was being created outside the closure - thus the FnMut capture outer variable error without `clone()`

moved the creation inside.